### PR TITLE
New example showing user download waveform.

### PR DESCRIPTION
**Pull request by yardov**
Hi reviewers, this is my first attempt to contribute. I wasn't able to do the "Signed-off-by: Name Lastname <name.lastmail@emailaddress.com>" but my name is in there. Is there I way to add this to a commit via VS Code?
Signed-off-by: Gerardo Orozco <gerardo.orozco@ni.com>

### What does this Pull Request accomplish?

This is a new RFSG example to show users how to create, normalize and download a user provided waveform. In this case, is multitone and all the code you need is included on this example.
This is specially important for users coming from Playback library. RFSG has now the functionality to do the same but the steps need to be follow correctly to get the exact power right. 

### Why should this Pull Request be merged?

It's an example.

### What testing has been done?

The algorithm is tested with multiple code documented test cases and the limitations are noted there.
The example is written following as close as possible the same guidelines of the other examples at the time of commit.

### CODE REVIEW

**Comment by reckenro** [comment]
Sorry, just now getting to this but you should be able to configure `black` locally to get the python formatted correctly. You can see this Wiki page we have for setting it up: https://github.com/ni/grpc-device/wiki/Development-Tips-and-Tricks#black-format-on-save

**Comment by bkeryan** [comment]
Note that `black` has configuration options in https://github.com/ni/grpc-device/blob/main/pyproject.toml

Running it locally will automatically use that configuration.

**Comment by yardov** [comment]
@bkeryan thank you, your pointes to the Black configuration helped.
I used: https://black.vercel.app/
I am not able to get it working with my VS Code for some reason.

Now that is passed there, can I assume you guys will give me comments on anything missing for it to make it to the main?

File: examples/nirfsg/rfsg-multitone-example.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+import sys
+
+import grpc
+import nidevice_pb2 as nidevice_grpc
+import nirfsg_pb2 as nirfsg_types
+import nirfsg_pb2_grpc as grpc_nirfsg
+import numpy as np
+
+
+SERVER_ADDRESS = "mercury07"
```
Comment (reckenro) [inline examples/nirfsg/rfsg-multitone-example.py:41]: Guessing this was for testing. Can you update this to "localhost"?

File: examples/nirfsg/rfsg-multitone-example.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    client.SetAttributeViString(
+        nirfsg_types.SetAttributeViStringRequest(
+            vi=vi,
+            channel_name="",
+            attribute_id=nirfsg_types.NIRFSG_ATTRIBUTE_SELECTED_PORTS,
+            value_raw=rfsg_selected_ports,
+        )
+    )
+
+    # We wante the tones to be at a specific power, for that reason, we adjust
```
Comment (reckenro) [inline examples/nirfsg/rfsg-multitone-example.py:179]: [Minor] typo "wante" -> "want"

File: examples/nirfsg/rfsg-multitone-example.py [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+    client.SetAttributeViReal64(
+        nirfsg_types.SetAttributeViReal64Request(
+            vi=vi,
+            channel_name="",
+            attribute_id=nirfsg_types.NIRFSG_ATTRIBUTE_FREQUENCY,
+            value_raw=center_frequency,
+        )
+    )
+
+    # Coarce Settings
```
Comment (reckenro) [inline examples/nirfsg/rfsg-multitone-example.py:209]: Coerce or Coarse I'm guessing.

**Comment by reckenro** [comment]
> Hi reviewers, this is my first attempt to contribute. I wasn't able to do the "Signed-off-by: Name Lastname [name.lastmail@emailaddress.com](mailto:name.lastmail@emailaddress.com)" but my name is in there. Is there I way to add this to a commit via VS Code?

Honestly, I don't think we've done this across the board and also Squash Merge before committing to main so I need to look into the policy more. But anyways, I think in VS Code you can do it with the three little dot menu next to the repo in the source control tab:

Commit signed off

I think there's also a more general setting in VS Code to enable it:

image

**Review by reckenro — APPROVED** [review]
Left a few minor comments but overall looks good to me

**Comment by reckenro** [comment]
> @bkeryan thank you, your pointes to the Black configuration helped. I used: https://black.vercel.app/ I am not able to get it working with my VS Code for some reason.
> 
> Now that is passed there, can I assume you guys will give me comments on anything missing for it to make it to the main?

Yes, I reviewed and added some more comments. And we have it set up pretty well with required Checks and policies so conversations have to be resolved and there has to be at least one approving reviewer.

The system tests (run_win_system_tests and run_ubuntu_system_tests) aren't currently marked required because some are intermittent but for this specific change those don't matter because they don't interact with the examples. Mainly just that validate_python step applies to the examples.

**Comment by bkeryan** [comment]
> > Hi reviewers, this is my first attempt to contribute. I wasn't able to do the "Signed-off-by: Name Lastname [name.lastmail@emailaddress.com](mailto:name.lastmail@emailaddress.com)" but my name is in there. Is there I way to add this to a commit via VS Code?
> 
> Honestly, I don't think we've done this across the board and also we Squash Merge before committing to main so I need to look into the policy more and see if we should enforce it more/less. But anyways, I think in VS Code you can do it with the three little dot menu next to the repo in the source control tab:
> 
> Commit signed off
> 
> I think there's also a more general setting in VS Code to enable it:
> 
> image

Sign-off is for compliance with a DCO (Developer Certificate of Origin) and it is different than commit signing (which uses GPG to cryptographically sign commits).

image

File: examples/nirfsg/rfsg-multitone-example.py (hunk 2 above)
Comment (yardov) [reply examples/nirfsg/rfsg-multitone-example.py]: implemented

File: examples/nirfsg/rfsg-multitone-example.py (hunk 3 above)
Comment (yardov) [reply examples/nirfsg/rfsg-multitone-example.py]: implemented

**Review by astarche — APPROVED** [review]

**Pull request merged**